src/Chainweb/Pact/PactService.hs

```diff
@@ -425,6 +429,7 @@ serviceRequests logFn memPoolAccess reqQ = do
                     , "): "
                     , show e
                     ]
+                cleanupDb
```

Comment:
This removes any possibly pending save points from the sqlite context. Generally, this should not be necessary, because all inner logic (in particular `withCheckpointer`) are supposed to do proper cleanup, even in case of exceptions.

However, code can have bugs and we've observed that it can actually happen that there are dangling SQL savepoints which leads to sqlite never committing anything to disk and results in memory leaks.

This change fixes the memory leak by allowing pact service to recover, but doesn't resolve the underlying issue.